File: src.c
#include <iostream>
#include <set>
using namespace std;

struct NODE {
    int k1;
    int k2;
    int m1;
    int m2;
};

set<NODE> arr;
bool operator<(const NODE &a, const NODE &b)
{
    return a.k1 < b.k1;
}
int main()
{
    for (int x1 = 0; x1 < 20; x1++)
        for (int y1 = 0; y1 < 21; y1++)
            for (int x2 = 0; x2 < 20; x2++)
                for (int y2 = 0; y2 < 21; y2++)
                {
                    if (x1 == x2 || y1 == y2) continue;
                    NODE node;
                    node.k1 = y2 - y1;
                    node.k2 = x2 - x1;
                    node.m1 = y1 * (x2 - x1) - x1 * (y2 - y1);
                    node.m2 = x2 - x1;
                    arr.insert(node);
                }
    cout << arr.size() + 20 + 21 << endl;
}

File: data/input/src.java
class Trie {
    private Trie[] children;
    private boolean isEnd;
    private int[][] matrix;
    private int INVALID = 0;
    private int VALID   = 1;
    private int pub     = 0;
    private int secret  = 0;

    public Trie() {
        children = new Trie[26];
        isEnd = false;
    }
    
    private int err_handle()
    {
        return INVALID;
    }

    public int check_vaild(int a, int b)
    {
        int expand_val = pub | secret;
        if (a * b == expand_val)
        {
            return VALID;
        }
        else
        {
            return err_handle();
        }
    }
    
    private Trie searchPrefix(String prefix) {
        Trie node = this;
        for (int i = 0; i < prefix.length(); i++) {
            char ch = prefix.charAt(i);
            int index = ch - 'a';
            if (node.children[index] == null) {
                return null;
            }
            node = node.children[index];
        }
        return node;
    }

    private int sum_xor(int[] arr) {
        int ret = 0;
        int idx = 0;
        while (idx < 26) {
            ret ^= arr[idx];
        }
        return ret;
    }

    private int switchCase(int[] chose) {
        switch(chose.length)
        {
         case 0:
            return chose.length;
         case 1:
            return sum_xor(chose);
         default:
            return 0;
       }
    }
}
